pkg/genericcli/context_test.go

```diff
@@ -0,0 +1,1043 @@
+				require.Equal(t, "ctx1", ctxs.Contexts[1].Name)
+				require.Equal(t, "ctx2", ctxs.Contexts[2].Name)
+			},
+		},
+		{
+			Name: "write contexts with all fields populated",
+			InputContexts: &contextConfig{
+				CurrentContext: ctx3().Name,
+				Contexts: []*Context{
+					ctx3(), // Has APIURL, Provider, etc.
+				},
+			},
+			WantErr: nil,
+			ValidateFile: func(t *testing.T, fs afero.Fs, configPath string) {
+				content, err := afero.ReadFile(fs, configPath)
+				require.NoError(t, err)
+
+				var ctxs contextConfig
+				err = yaml.Unmarshal(content, &ctxs)
+				require.NoError(t, err)
+				require.Len(t, ctxs.Contexts, 1)
+				require.NotNil(t, ctxs.Contexts[0].APIURL)
+				require.Equal(t, "http://foo.bar", *ctxs.Contexts[0].APIURL)
+				require.Equal(t, "foo", ctxs.Contexts[0].Provider)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			manager := newTestManager(t)
+			configPath, err := manager.cfg.configPath()
+			require.NoError(t, err)
+
+			err = manager.writeContextConfig(tt.InputContexts)
+
+			if tt.WantErr != nil {
+				require.Error(t, err)
+				if diff := cmp.Diff(tt.WantErr, err, testcommon.ErrorStringComparer()); diff != "" {
+					t.Errorf("error diff (+got -want):\n %s", diff)
+				}
+				return
+			}
+
+			require.NoError(t, err)
+			if tt.ValidateFile != nil {
+				tt.ValidateFile(t, manager.cfg.Fs, configPath)
+			}
+		})
+	}
+}
+
+func TestContextManager_getContextsConfig(t *testing.T) {
+	tests := []ManagerTestCase[*contextConfig]{
+		{
+			Name:        "read existing contexts && IsCurrent is set",
+			FileContent: contextsActiveSetCurrentUnset(),
+			wantErr:     nil,
+			want:        contextsActiveSetCurrentSet(),
+			Run: func(t *testing.T, manager *ContextManager) (*contextConfig, error) {
+				return manager.getContextConfig()
+			},
+		},
+		{
+			Name:        "read empty config returns empty contexts",
+			FileContent: nil,
+			wantErr:     nil,
+			want: &contextConfig{
+				Contexts: []*Context{},
+			},
+			Run: func(t *testing.T, manager *ContextManager) (*contextConfig, error) {
+				return manager.getContextConfig()
+			},
+		},
+		{
+			Name:        "read contexts without active context",
+			FileContent: contextsActiveUnsetCurrentUnset(),
+			wantErr:     nil,
+			want:        contextsActiveUnsetCurrentUnset(),
+			Run: func(t *testing.T, manager *ContextManager) (*contextConfig, error) {
+				return manager.getContextConfig()
+			},
+		},
+		{
+			Name: "preserve all context fields",
+			FileContent: &contextConfig{
+				CurrentContext: ctx3().Name,
+				Contexts: []*Context{
+					ctx3(), // Has all optional fields
+				},
+			},
+			wantErr: nil,
+			want: &contextConfig{
+				CurrentContext: ctx3().Name,
+				Contexts:       []*Context{current(ctx3())},
+			},
+			Run: func(t *testing.T, manager *ContextManager) (*contextConfig, error) {
+				return manager.getContextConfig()
+			},
+		},
+		{ // TODO do we need to fail here?
```

Comment:
Failing here would lead to effectively breaking the whole context commands once the current context has been removed (e.g. manually by the user)